app/scripts/models/user.js

```diff
@@ -222,6 +222,30 @@ define([
       return (Session.email && Session.sessionToken &&
         (! Session.cachedCredentials ||
         Session.cachedCredentials.email !== Session.email));
+    },
+
+    signInAccount: function (account, relier) {
+      var self = this;
+      return account.signIn(relier)
+        .then(function () {
+          var oldAccount = self.getAccountByUid(account.get('uid'));
+          if (! oldAccount.isDefault()) {
+            // allow new account attributes to override old ones
+            oldAccount.set(_.omit(account.attributes, function (val) {
```

Comment:
Can you add a note here that this is what allows permissions to remain persisted per relier across sign ins?